src/process/process.cr

```diff
@@ -88,14 +112,9 @@ class Process
     pid
   end
 
+  # :nodoc:
   def self.waitpid pid
-    chan = Event::SignalChildHandler.instance.waitpid pid
-    status = chan.receive
-    if status
-      return status
-    else
-      raise "waitpid channel closed"
-    end
+    {{ raise "DEPRECATED: fork and Process.new return a Process.  Use process.wait instead." }}
   end
```

Comment:
I think if we drop this and don't let `Process.kill` accept pids anymore, we should also drop `Process.kill` and make the primary interface `Process#kill` instead.